airbyte-cdk/python/airbyte_cdk/sources/file_based/config/avro_format.py

```diff
@@ -10,8 +10,8 @@ class AvroFormat(BaseModel):
     filetype: Literal["avro"] = "avro"
 
     # This option is not recommended, but necessary for backwards compatibility
-    decimal_as_float: bool = Field(
-        title="Convert Decimal Fields to Floats",
-        description="Whether to convert decimal fields to floats. There is a loss of precision when converting decimals to floats, so this is not recommended.",
-        default=False,
+    double_as_string: bool = Field(
+        title="Convert Double Fields to Strings",
+        description="Whether to convert double fields to strings. There is a loss of precision when converting decimals to floats, so this is recommended.",
+        default=True,
```

Comment:
And to confirm. For legacy S3 connectors, the adapter will by default set this to False if I understand correct.

Reply:
yes, created an issue https://github.com/airbytehq/airbyte/issues/29225